theano/gof/op.py

```diff
@@ -1377,7 +1436,12 @@ def get_op_params(self):
         The names must be strings that are not a C keyword and the
         values must be strings of literal C representations.
 
+        If your COp defines an attribute `enum` created from ``enum_list()``
+        or ``enum_type()``, then this method will try to deduce macros from
+        the enum fields.
         """
+        if hasattr(self, 'enum') and hasattr(self.enum, '__dict__'):
+            return [(k, str(v)) for (k, v) in self.enum.__dict__.items()]
```

Comment:
Sorry, I had forgotten that using `__dict__` works correctly only if enum is a `namedtuple` (in that case, a_named_tuple.__dict__ contains only attributes we define). Indeed, this code is unsafe.